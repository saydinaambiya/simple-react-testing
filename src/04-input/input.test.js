import {render, screen} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Input from "./input";

describe("Input component", () => {
    test("should display a label and input with empty value", () => {
        render(<Input/>)

        const labelElement = screen.getByTestId("input-label");
        const inputElement = screen.getByTestId("input-username");

        expect(labelElement.textContent).toEqual("");
        expect(inputElement.value).toEqual("");
    })

    test("should update h3 when user inpu text", () => {
        const {asFragment} = render(<Input/>)

        const labelElement = screen.getByTestId("input-label");
        const inputElement = screen.getByTestId("input-username");

        userEvent.type(inputElement, "John");

        expect(asFragment()).toMatchSnapshot();

        expect(labelElement.textContent).toEqual("John");
        expect(inputElement.value).toEqual("John");
    })

})
