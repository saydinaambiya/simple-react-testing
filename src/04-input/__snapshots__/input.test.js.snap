// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should update h3 when user inpu text 1`] = `
<DocumentFragment>
  <h3
    data-testid="input-label"
  >
    John
  </h3>
  <input
    data-testid="input-username"
    value="John"
  />
</DocumentFragment>
`;
